test/functional/blocknotify.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test blocknotify.
+
+Verify that a bitcoind node calls the given command for each block
+"""
+
+import os
+import time
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class BlockNotifyTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.path = os.path.join(self.options.tmpdir, 'blocks.txt')
+
+        # the command will append the notified block hash to file
+        self.extra_args = [['-blocknotify=echo %%s >> %s' % self.path]]
```

Comment:
Maybe escape path with `shlex.quote(self.path)`

Reply:
`tmpdir` is used elsewhere (should be secure) and `blocks.txt` doesn't need to be quoted. Is there a strong argument for it?

Reply:
Of you mean to quote the command `echo %%s >> %s`?

Reply:
Quoting the entire command would be incorrect because it is passed literally to [system(3)](http://man7.org/linux/man-pages/man3/system.3.html). Quoting the path is correct because it is being passed as a shell token. Anyway just suggested this in case you wanted the code to be more correct. There are no practical implications (security or otherwise) either way.

Reply:
Right, otherwise it wouldn't work 😃 I'm going for simplicity if you don't mind.